### PR TITLE
Move specs in resource module

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

Move the `qml.specs` transform to the resource module, update the documentation and move the tests as well.

### CODE REVIEW

File: pennylane/resource/__init__.py [hunk 1]
```diff
@@ -27,6 +27,16 @@
...
 
+Circuit specifications
+----------------------
+
+.. currentmodule:: pennylane.resource
+
+.. autosummary::
+    :toctree: api
+
+    ~specs
```
Comment (trbromley) [inline pennylane/resource/__init__.py:38]:
```suggestion
    ~pennylane.specs
```
This will allow us to go with the top level one

Comment (rmoyard) [reply pennylane/resource/__init__.py]: I don't think it works:
`WARNING: [autosummary] failed to import 'pennylane.resource.pennylane.specs': no module named pennylane.resource.pennylane.specs`

File: pennylane/resource/__init__.py [hunk 2]
```diff
@@ -27,6 +27,16 @@
...
 
+Circuit specifications
+----------------------
+
+.. currentmodule:: pennylane.resource
+
+.. autosummary::
+    :toctree: api
+
+    ~pennylane.specs
```
Comment (rmoyard) [inline pennylane/resource/__init__.py:38]:
```suggestion
    ~specs

```

File: pennylane/resource/__init__.py (hunk 1 above)
Comment (josh146) [reply pennylane/resource/__init__.py:38]: @rmoyard the key here is to use `.. currentmodule::` to set the correct relative location

**Comment by josh146** [comment]
@rmoyard can you re-tag me for review once the specs autosummary location has been fixed :)

File: pennylane/resource/__init__.py (hunk 1 above)
Comment (rmoyard) [reply pennylane/resource/__init__.py:38]: Thanks !

**Review by trbromley — APPROVED** [review]
Thanks @rmoyard! CC @soranjh for this change.

**Review by josh146 — APPROVED** [review]
thanks @rmoyard!

**Pull request merged**